src/client_input_handler.rs

```diff
@@ -240,24 +238,67 @@ fn get_current_gamestate() -> Msg {
 fn wait_for_my_turn() -> Msg {
     let two_sec = time::Duration::from_secs(2);
     thread::sleep(two_sec);
-    return get_current_gamestate();
+    get_current_gamestate()
+}
+
+fn check_is_move_valid(move_to_make: usize, game_state: &GameState, am_i_player_one: bool) -> bool {
+    let range_of_valid_moves = if am_i_player_one {
+        Range {
+            start: 1,
+            end: SLOTS,
+        }
+    } else {
+        Range {
+            start: SLOTS + 1,
+            end: BOARD_LENGTH,
+        }
+    };
+    if range_of_valid_moves.contains(&move_to_make) && game_state.game_board[move_to_make] != 0 {
+        return true;
+    }
+    if !range_of_valid_moves.contains(&move_to_make) {
+        println!(
+            "Move was not in the playable range of moves, which is {:?} for you.",
+            &range_of_valid_moves
```

Comment:
This range is incorrect. Player1 should be able to move slots 1-6 but the message gives the range `1..7` and player2 should be 8-13 but error says `8..14`

Reply:
ah yes, the range object is [start, end) so the end is not actually included...

Reply:
updated